libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -577,9 +570,8 @@ uint16_t AP_Logger_File::find_last_log()
     if (fname == nullptr) {
         return ret;
     }
-    hal.scheduler->expect_delay_ms(3000);
+    EXPECT_DELAY(hal, 3000);
     int fd = open(fname, O_RDONLY|O_CLOEXEC);
-    hal.scheduler->expect_delay_ms(0);
```

Comment:
This is a behaviour change as the delay object will remain in scope well past the point we were zeroing the delay here.

Given this one covers reading as well, this one is more serious than the others...

Reply:
I am happy with the wider scope